pkg/cmd/cli/cmd/expose.go

```diff
@@ -46,6 +51,14 @@ func validate(cmd *cobra.Command, f *clientcmd.Factory, args []string) error {
 	if cmdutil.GetFlagString(cmd, "generator") != "route/v1" {
 		return nil
 	}
+	// We don't want to use any other name for our route other than
+	// the one the service we expose has
+	providedName := cmdutil.GetFlagString(cmd, "name")
```

Comment:
If we can add our own flags on top then it's going to be trivial to add parameters in the generator. Gonna play around with it on Monday.